lib/buffer.js

```diff
@@ -158,18 +157,16 @@ function assertSize(size) {
  **/
 Buffer.alloc = function(size, fill, encoding) {
   assertSize(size);
-  if (size <= 0)
-    return createBuffer(size);
-  if (fill !== undefined) {
+  if (size > 0 && fill !== undefined) {
```

Comment:
Hm, I think that would require benchmarking – `createUnsafeBuffer()` returns a slice from the pool, so I’d actually expect that to be faster than an extra typed array allocation.